aiidalab_widgets_base/computers.py

```diff
@@ -0,0 +1,718 @@
+            return self._inp_proxy_address.value
+
+    @proxy_hostname.setter
+    def proxy_hostname(self, proxy_hostname):
+        self._use_proxy.value = True
+        self._inp_proxy_address.value = proxy_hostname
+
+    @property
+    def proxy_username(self):
+        if len(self._inp_proxy_username.value.strip()) == 0:
+            return None
+        else:
+            return self._inp_proxy_username.value
+
+    @proxy_username.setter
+    def proxy_username(self, proxy_username):
+        self._use_proxy.value = True
+        self._use_diff_proxy_username.value = True
+        self._inp_proxy_username.value = proxy_username
+
+class AiidaComputerSetup(ipw.VBox):
+    def __init__(self, **kwargs):
+        from aiida.transport import Transport
+        from aiida.scheduler import Scheduler
+        style = {"description_width":"200px"}
+
+        # list of widgets to be displayed
+        self._btn_setup_comp = ipw.Button(description="setup computer")
+        self._btn_setup_comp.on_click(self._on_setup_computer)
+        self._inp_computer_name = ipw.Text(value='',
+                                           placeholder='Will be only used within AiiDA',
+                                           description="AiiDA computer name:",
+                                           layout=ipw.Layout(width="500px"),
+                                           style=style)
+        self._computer_hostname = ipw.Dropdown(description="Select among configured hosts:",
+                                               layout=ipw.Layout(width="500px"),
+                                               style=style)
+        self._btn_refresh_computers = ipw.Button(description="refresh", layout={'margin': '0px 0px 20px 21s0px'})
+        self._btn_refresh_computers.on_click(self._get_available_computers)
+        self._inp_computer_description = ipw.Text(value='',
+                                                  placeholder='No description (yet)',
+                                                  description="Computer description:",
+                                                  layout=ipw.Layout(width="500px"),
+                                                  style=style)
+        self._computer_workdir = ipw.Text(value='/scratch/{username}/aiida_run',
+                                          description="Workdir:",
+                                          layout=ipw.Layout(width="500px"),
+                                          style=style)
+        self._computer_mpirun_cmd = ipw.Text(value='mpirun -n {tot_num_mpiprocs}',
+                                             description="Mpirun command:",
+                                             layout=ipw.Layout(width="500px"),
+                                             style=style)
+        self._computer_ncpus = ipw.IntText(value=12,
+                                           step=1,
+                                           description='Number of CPU(s) per node:',
```

Comment:
No parenthesis needed

Reply:
why? it can also be 1 cpu...

Reply:
you would still say "number of central processing units"

Reply:
Ok